### DISCUSSION TITLE
Is it possible to connect to Azure Cosmos DB in direct mode?

### DISCUSSION THREAD

**Discussion by donruperto** [Gremlinq/ExRam.Gremlinq#1793] [Q&A] [answered]
Hi,

is it possible to connect to Microsoft Azure Cosmos DB in [direct](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.connectionmode?view=azure-dotnet) mode, like you can do with Azure Cosmos DB Core SDK? As far as I can see in AppInsights the connection is established in Gateway Mode.

Thx

Rupert

**Comment by danielcweber** [comment, marked as answer]
I wasn't aware of different connection modes and apparently, it's [not applicable to the Gremlin API](https://github.com/MicrosoftDocs/azure-docs/issues/48215).
